.github/workflows/test-pull_request.yaml

```diff
@@ -156,3 +156,74 @@ jobs:
           else
               "./bin/$CONFIGURATION/unittests" --gtest_shuffle
           fi
+
+  build_mac:
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+        arch: [x86_64]
+    name: Mac
+    runs-on: macos-latest
+    steps:
+      - name: Cache Qt
+        id: cache-qt-lin
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/../Qt
+          key: ${{ runner.os }}-QtCache-${{ env.QT_VERSION }}
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v2
+        with:
+          version: ${{ env.QT_VERSION }}
+          dir: ${{ github.workspace }}/..
+          cached: ${{ steps.cache-qt-lin.outputs.cache-hit }}
+          setup-python: 'false'
+          aqtversion: ==1.1.3
+
+      - uses: actions/checkout@v1
+        name: Checkout
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ runner.os }}-${{ matrix.configuration }}-${{ matrix.compiler }}
+
+      - name: Configure CMake
+        env:
+          CONFIGURATION: ${{ matrix.configuration }}
+          COMPILER: ${{ matrix.compiler }}
+          ARCHITECTURE: ${{ matrix.arch }}
+          JOB_CMAKE_OPTIONS: ${{ matrix.cmake_options }}
+        run: |
+          mkdir build
+          cd build
+          
+          export CXXFLAGS="-mtune=generic -pipe -Wno-unknown-pragmas"
+          export CFLAGS="-mtune=generic -pipe -Wno-unknown-pragmas"
+          export CMAKE_OSX_ARCHITECTURES="$ARCHITECTURE"
+
+          brew install ninja
+          cmake -G Ninja -DFSO_FATAL_WARNINGS=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON $CMAKE_OPTIONS \
+          -DCMAKE_INSTALL_PREFIX="$(pwd)/install" -DCMAKE_BUILD_TYPE=$CONFIGURATION \
+          -DFFMPEG_USE_PRECOMPILED=ON -DFSO_BUILD_TESTS=ON -DFSO_BUILD_INCLUDED_LIBS=ON -DFSO_BUILD_QTFRED=ON \
+          -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+          -DSHADERS_ENABLE_COMPILATION=ON ..
+      - name: Compile
+        working-directory: ./build
+        run: LD_LIBRARY_PATH=$Qt5_DIR/lib:$LD_LIBRARY_PATH ninja all
```

Comment:
Missing ninja argument `-k 20`?

Reply:
Why do we have `-k 20` again? Looks like it'd ignore errors with this setting.

Reply:
`-k 20` is probably there to re-try building the entire thing up to 20 times.  We do have external dependencies and there are some projects/files within the build chain that are generated code that will fail the process the first time around.  We can try omitting this argument to see what it does in the CI.

Reply:
The Ninja docs say this: `-k     keep going until N jobs fail (0 means infinity) [default=1]`. It looks like it just makes it try to build all targets even if there are errors, instead of failing early. I remember it was making debugging the build script difficult, so I removed it. I can bring it back, just wondering what it was for.

Reply:
If its building correctly in your tests then we can keep it omitted until further notice

Reply:
Yeah it works fine.